content/en/docs/develop/porting.md

```diff
@@ -541,3 +535,294 @@ When this option is enabled, we can either:
    }
    ```
 
+## Relevant files to the build tool
+
+In order for an application or library to work with Unikraft, you need to provide some specific files:
+
+1. **Makefile**:
+   Used to specify where the main Unikraft repository is with respect to the application's repo, as well as repos for any external libraries the application needs to use.
+   The `Makefile` is not required when porting an external library.
+
+1. **Makefile.uk**:
+   The main Makefile used to specify which sources to build (and optionally where to fetch them from), include paths, flags and any application-specific targets.
+
+1. **Config.uk**:
+   A [Kconfig-like](https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html) snippet used to populate Unikraft's menu with application-specific options.
+
+1. **exportsyms.uk**:
+   A text file where each line contains the name of one symbol that should be exported to other libraries.
+   This file usually contains only `main` for an application that is developed/ported as a single library to Unikraft.
+   If an `exportsyms.uk` file is not present, all symbols will be exported.
+
+1. **extra.ld**:
+   [Optional] Contains an amendment to the main linker script.
+
+In the tutorial below, we will focus on files written for an application.
+The process of porting a library is very similar, just change the `APP` prefix of the variable names to `LIB`.
+
+### Makefile
+
+The Makefile is generally short and simple and might remind you to Linux kernel modules that are built off-tree.
+For most applications the Makefile should contain no more than the following:
+
+```Makefile
+UK_ROOT  ?= $(PWD)/../../unikraft
+UK_LIBS  ?= $(PWD)/../../libs
+UK_PLATS ?= $(PWD)/../../plats
+LIBS  := $(UK_LIBS)/lib1:$(UK_LIBS)/lib2:$(UK_LIBS)/libN
+PLATS ?=
+
+all:
+        @make -C $(UK_ROOT) A=$(PWD) L=$(LIBS) P=$(PLATS)
+
+$(MAKECMDGOALS):
+        @make -C $(UK_ROOT) A=$(PWD) L=$(LIBS) P=$(PLATS) $(MAKECMDGOALS)
+```
```

Comment:
Mention that the `Makefile` (as a reader may notice) is just a wrapper around the [core `Makefile`](https://github.com/unikraft/unikraft/blob/staging/Makefile).

You could get the same result by using the command:

```
make -C ../../unikraft A=. L=../../libs/lib1:../../libs/lib2:../../libs/libN
```